dhcp-controller/controllers/dhcpserver_controller.go

```diff
@@ -173,114 +191,225 @@ func (r *DHCPServerReconciler) ensureServer(request reconcile.Request,
 				return &reconcile.Result{}, err
 			}
 			log.Info("server spec is changed, updating ConfigMap")
-			// Delete the deployment, will be recreated with new configmap
-			err = r.Delete(context.TODO(), dep)
+			// Delete the VM, will be recreated with new configmap
+			err = r.Delete(context.TODO(), vm)
 			if err != nil {
-				log.Error(err, "Failed to delete the deployment or there is none")
+				log.Error(err, "Failed to delete the VM or there is none")
 			}
-			err = r.Get(context.TODO(), types.NamespacedName{Name: dep.Name, Namespace: server.Namespace}, found)
+			err = r.Get(context.TODO(), types.NamespacedName{Name: server.Name, Namespace: server.Namespace}, found)
 			for {
 				if err != nil && errors.IsNotFound(err) {
 					break
 				}
-				err = r.Get(context.TODO(), types.NamespacedName{Name: dep.Name, Namespace: server.Namespace}, found)
+				err = r.Get(context.TODO(), types.NamespacedName{Name: server.Name, Namespace: server.Namespace}, found)
 				time.Sleep(2 * time.Second)
 			}
 		} else {
 			log.Info("server spec is unchanged, not updating ConfigMap")
 		}
 	}
-	// See if deployment already exists and create if it doesn't
+	// See if VM already exists and create if it doesn't
 	err = r.Get(context.TODO(), types.NamespacedName{
-		Name:      dep.Name,
+		Name:      server.Name,
 		Namespace: server.Namespace,
 	}, found)
 	if err != nil && errors.IsNotFound(err) {
 
-		// Create the deployment
-		err = r.Create(context.TODO(), dep)
+		// Create the VM
+		err = r.Create(context.TODO(), vm)
 
 		if err != nil {
-			// Deployment failed
+			// VM failed
 			return &reconcile.Result{}, err
 		} else {
-			// Deployment was successful
+			// VM was successful
 			return nil, nil
 		}
 	} else if err != nil {
-		// Error that isn't due to the deployment not existing
+		// Error that isn't due to the VM not existing
 		return &reconcile.Result{}, err
 	}
 
 	return nil, nil
 }
 
-// backendDeployment is a code for Creating Deployment
-func (r *DHCPServerReconciler) backendDeployment(v dhcpv1alpha1.DHCPServer) *appsv1.Deployment {
+// backendVM is a code for Creating VM
+func (r *DHCPServerReconciler) backendVM(v dhcpv1alpha1.DHCPServer) *unstructured.Unstructured {
```

Comment:
creating a template file and rendering that will make it more readable and easy to maintain.